### PR TITLE
added rouge and made separate component for raw_code

### PR DESCRIPTION
**Pull request by Haumer**
## Context
- moving away from embedded github scripts to displaying the code of gists server side

## Changes
- [x] added [rouge](https://github.com/rouge-ruby/rouge) / [rouge-rails](https://github.com/jacobsimeon/rouge-rails) 
using this as a guide 👉  [https://stackoverflow.com/questions/5009031/file-open-write-and-save](https://stackoverflow.com/questions/5009031/file-open-write-and-save)
- [x] syntax highlighting works

## Example

Screenshot 2020-11-07 at 12 52 32

### CODE REVIEW

File: app/components/gist/files_component.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Gist::FilesComponent < ViewComponent::Base
+  attr_reader :gist
+
```
Comment (Haumer) [inline app/components/gist/files_component.rb:3]: view component doctrine suggests to move the attr_readers to the bottom under a `private` tag so it cant be accessed from outside the component 🙈

Comment (barangerbenjamin) [reply app/components/gist/files_component.rb]: God that's Scheisse

File: app/components/gist/files_component.rb [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
 
   def initialize(gist:)
     @gist = gist
   end
+
+  private
+
+  def language(name)
+
+  end
```
Comment (barangerbenjamin) [inline app/components/gist/files_component.rb:12]: Is that a component thing?

File: app/components/gist/files_component.html.erb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <% gist.gist_files.each do |file| %>
   <div class="highlight">
-   <%= helpers.rouge(file.raw_code, "python").html_safe %>
+   <%= rouge(file.raw_code, file.language.rouge_name).html_safe %>
```
Comment (barangerbenjamin) [inline app/components/gist/files_component.html.erb:3]: this fail on my side, I get undefined method rouge. that's why I was fetching it from helpers 😢

Comment (Haumer) [reply app/components/gist/files_component.html.erb]: view components are encapsulated so dont have access to helpers. i included the application helper in the viewcomponent.rb file so all good!

File: app/components/gist/files_component.rb (hunk 2 above)
Comment (Haumer) [reply app/components/gist/files_component.rb:12]: no thats just a derp

**Pull request merged**